napalm_ansible/modules/napalm_get_config.py

```diff
@@ -0,0 +1,274 @@
+"""
+(c) 2018 Dmitry Zykov https://dmitryzykov.com
+    Based on module napalm_install_conf by Elisa Jasinska <elisa@bigwaveit.org>
+    Original prototype by David Barroso <dbarrosop@dravetech.com>
+
+This file is part of Ansible
+
+Ansible is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Ansible is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
+import os.path
+import hashlib
+
+from ansible.module_utils.basic import AnsibleModule, return_values
+
+DOCUMENTATION = '''
+---
+module: napalm_get_config
+author: "Dmitry Zykov (@dmitryzykov)"
+version_added: "1.1"
+short_description: "Get and save to file config taken from a device supported by NAPALM"
+description:
+    - "This module will get configuration from a device with any
+       OS supported by napalm and save it to a file."
+requirements:
+    - napalm
+options:
+    hostname:
+        description:
+          - IP or FQDN of the device you want to connect to
+        required: False
+    username:
+        description:
+          - Username
+        required: False
+    password:
+        description:
+          - Password
+        required: False
+    provider:
+        description:
+          - Dictionary which acts as a collection of arguments used to define the characteristics
+            of how to connect to the device.
+            Note - hostname, username, password and dev_os must be defined in either provider
+            or local param
+            Note - local param takes precedence, e.g. hostname is preferred to provider['hostname']
+        required: False
+    dev_os:
+        description:
+          - OS of the device
+        required: False
+        choices: ['eos', 'junos', 'iosxr', 'fortios', 'ios', 'mock', 'nxos', 'nxos_ssh', 'panos',
+        'vyos']
+    timeout:
+        description:
+          - Time in seconds to wait for the device to respond
+        required: False
+        default: 60
+    optional_args:
+        description:
+          - Dictionary of additional arguments passed to underlying driver
+        required: False
+        default: None
+    dest:
```

Comment:
I think it might be valuable to make this optional instead and let users download the config without having it to save into a file.

Reply:
done, "dest" argument is optional now